relay-server/src/services/processor/ourlog.rs

```diff
@@ -112,13 +128,54 @@ fn scrub(
     Ok(())
 }
 
+#[cfg(any(feature = "processing", test))]
+fn process_attribute_types(ourlog: &mut OurLog) {
+    let Some(attributes) = ourlog.attributes.value_mut() else {
+        return;
+    };
+
+    let attributes = attributes.iter_mut().map(|(_, attr)| attr);
+
+    for attribute in attributes {
+        use OurLogAttributeType::*;
+
+        let Some(inner) = attribute.value_mut() else {
+            continue;
+        };
+
+        match (&mut inner.value.ty, &mut inner.value.value) {
+            (Annotated(Some(Boolean), _), Annotated(Some(Value::Bool(_)), _)) => (),
+            (Annotated(Some(Integer), _), Annotated(Some(Value::I64(_)), _)) => (),
+            (Annotated(Some(Integer), _), Annotated(Some(Value::U64(_)), _)) => (),
+            (Annotated(Some(Double), _), Annotated(Some(Value::I64(_)), _)) => (),
+            (Annotated(Some(Double), _), Annotated(Some(Value::U64(_)), _)) => (),
+            (Annotated(Some(Double), _), Annotated(Some(Value::F64(_)), _)) => (),
+            (Annotated(Some(String), _), Annotated(Some(Value::String(_)), _)) => (),
+            (Annotated(ty_opt @ Some(Unknown(_)), ty_meta), _) => {
+                ty_meta.add_error(ErrorKind::InvalidData);
+                ty_meta.set_original_value(ty_opt.take());
+            }
+            (Annotated(Some(_), _), Annotated(value @ Some(_), value_meta)) => {
+                value_meta.add_error(ErrorKind::InvalidData);
+                value_meta.set_original_value(value.take());
+            }
+            (Annotated(None, _), _) | (_, Annotated(None, _)) => {
+                let original = attribute.value_mut().take();
+                attribute.meta_mut().add_error(ErrorKind::MissingAttribute);
+                attribute.meta_mut().set_original_value(original);
+            }
```

Comment:
This part should normally be already done by the `SchemaProcessor` which we definitely should enable for logs, similar to the `RemoveOther` mentioned elsewhere.